packages/x-tree-view/src/internals/plugins/useTreeViewKeyboardNavigation/useTreeViewKeyboardNavigation.ts

```diff
@@ -89,7 +89,10 @@ export const useTreeViewKeyboardNavigation: TreeViewPlugin<
       return;
     }
 
-    if (event.altKey || event.currentTarget !== event.target) {
+    if (
```

Comment:
This ensure that the keyboard navigation works when the focus is on an element inside the item (most likely the checkbox added in this PR).

The previous check was here so that when you press a key, only the `handleKeyDown` of the item you press it is actually applied, not the one of its ancestors.
The new version keeps this behavior but improves it.

Reply:
Can the focus ever be on the `checkbox` if it is set to `tabIndex="-1"` and after clicking it the focus is set on the tree item? 🤔 
I'm just asking without debugging. 🙈

Reply:
Ok, that's fair point.